phpBB/phpbb/install/module/update_database/task/update.php

```diff
@@ -0,0 +1,176 @@
+
+use phpbb\db\html_migrator_output_handler;
+use phpbb\db\log_wrapper_migrator_output_handler;
+use phpbb\db\migration\exception;
+use phpbb\install\exception\resource_limit_reached_exception;
+use phpbb\install\exception\user_interaction_required_exception;
+use phpbb\install\task_base;
+
+/**
+ * Database updater task
+ */
+class update extends task_base
+{
+	/**
+	 * @var \phpbb\config\config
+	 */
+	protected $config;
+
+	/**
+	 * @var \phpbb\extension\manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * @var \phpbb\filesystem\filesystem
+	 */
+	protected $filesystem;
+
+	/**
+	 * @var \phpbb\install\helper\config
+	 */
+	protected $installer_config;
+
+	/**
+	 * @var \phpbb\install\helper\iohandler\iohandler_interface
+	 */
+	protected $iohandler;
+
+	/**
+	 * @var \phpbb\language\language
+	 */
+	protected $language;
+
+	/**
+	 * @var \phpbb\log\log
+	 */
+	protected $log;
+
+	/**
+	 * @var \phpbb\db\migrator
+	 */
+	protected $migrator;
+
+	/**
+	 * @var \phpbb\user
+	 */
+	protected $user;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	//
+	// @todo: constructor
+	//
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function run()
+	{
+		// @todo: investigate hooks and stuff
+
+		$this->language->add_lang('migrator');
+
+		if (!isset($this->config['version_update_from']))
+		{
+			$this->config->set('version_update_from', $this->config['version']);
+		}
+
+		$original_version = $this->config['version_update_from'];
+
+		$this->migrator->set_output_handler(
+			new log_wrapper_migrator_output_handler(
+				$this->language,
+				new html_migrator_output_handler($this->language),
+				$this->phpbb_root_path . 'store/migrations_' . time() . '.log',
+				$this->filesystem
+			)
+		);
+
+		$this->migrator->create_migrations_table();
+
+		$migrations = $this->extension_manager
+			->get_finder()
+			->core_path('phpbb/db/migration/data/')
+			->extension_directory('/migrations')
+			->get_classes();
+
+		$this->migrator->set_migrations($migrations);
+
+		while (!$this->migrator->finished())
+		{
+			try
+			{
+				$this->migrator->update();
+			}
+			catch (exception $e)
+			{
+				$msg = $e->getParameters();
+				array_unshift($msg, $e->getMessage());
+
+				$this->iohandler->add_error_message($msg);
+				throw new user_interaction_required_exception();
+			}
+
+			if ($this->installer_config->get_time_remaining() <= 0 || $this->installer_config->get_memory_remaining() <= 0)
+			{
+				throw new resource_limit_reached_exception();
+			}
+		}
+
+		if ($original_version !== $this->config['version'])
+		{
+			$this->log->add(
+				'admin',
+				$this->user->data['user_id'],
+				$this->user->ip,
+				'LOG_UPDATE_DATABASE',
+				false,
+				array(
+					$original_version,
+					$this->config['version']
+				)
+			);
+		}
+
+		$this->iohandler->add_success_message('INLINE_UPDATE_SUCCESSFUL');
+
+		$this->config->delete('version_update_from');
+
+		// @todo: investigate phpbb_end_update()
```

Comment:
I was only wondering why you would want to do that after each time you run out of memory or time